qfieldsync/core/cloud_api.py

```diff
@@ -182,19 +264,33 @@ def server_urls() -> List[str]:
             "https://localhost:8002/",
         ]
 
+    def get_username(self) -> Optional[str]:
+        if self.auth_method == CloudAuthMethod.CREDENTIALS:
+            return self.auth().config("username")
+        elif self.auth_method == CloudAuthMethod.SSO:
+            return self.current_username
+        return None
+
     def auth(self) -> QgsAuthMethodConfig:
         auth_manager = QgsApplication.authManager()
 
         if not auth_manager.masterPasswordHashInDatabase():
             return QgsAuthMethodConfig()
 
         authcfg = self.preferences.value("qfieldCloudAuthcfg")
+
+        if not authcfg or authcfg == "":
```

Comment:
```suggestion
        if not authcfg:
```
Aren't they almost the same? Yes, now you also work with support authconfig to be `0` (int), but I don't think we should support that either.